engines/agos/metaengine.cpp

```diff
@@ -207,6 +213,197 @@ SaveStateList AgosMetaEngine::listSaves(const char *target) const {
+		gameId == "simon2") {
+
+		act = new Action("TXTFAST", _("Text speed - Fast"));
+		act->setCustomEngineActionEvent(AGOSAction::kActionTextSpeedFast);
+		act->addDefaultInputMapping("F1");
+		gameKeyMap->addAction(act);
+
+		act = new Action("TXTMEDIUM", _("Text speed - Medium"));
+		act->setCustomEngineActionEvent(AGOSAction::kActionTextSpeedMedium);
+		act->addDefaultInputMapping("F2");
+		gameKeyMap->addAction(act);
+
+		act = new Action("TXTSLOW", _("Text speed - Slow"));
+		act->setCustomEngineActionEvent(AGOSAction::kActionTextSpeedSlow);
+		act->addDefaultInputMapping("F3");
+		gameKeyMap->addAction(act);
+
+		act = new Action("SHOWOBJINTERACT", _("Show objects to interact"));
+		act->setCustomEngineActionEvent(AGOSAction::kActionShowObjects);
+		act->addDefaultInputMapping("F10");
+		gameKeyMap->addAction(act);
+
+		if (gameId == "simon2") {
+			act = new Action("BACKGRNDSND", _("Toggle background sounds On/Off"));
+			act->setCustomEngineActionEvent(AGOSAction::kActionToggleBackgroundSound);
+			act->addDefaultInputMapping("b");
+			gameKeyMap->addAction(act);
+		}
+	}
+
+	if (gameId == "feeble") {
+		act = new Action("SWTCHCHARACTER", _("Switch characters"));
+		act->setCustomEngineActionEvent(AGOSAction::kActionToggleSwitchCharacter);
+		act->addDefaultInputMapping("F7");
+		act->addDefaultInputMapping("JOY_X");
+		gameKeyMap->addAction(act);
+
+		act = new Action("TOGGLEHITBOX", _("Toggle hitbox names on/off"));
+		act->setCustomEngineActionEvent(AGOSAction::kActionToggleHitboxName);
+		act->addDefaultInputMapping("F9");
+		gameKeyMap->addAction(act);
+	}
+
+	if (gameId == "feeble" ||
+		gameId == "simon2") {
+		act = new Action("TOGGLESUB", _("Switches between speech only and combined speech and subtitles"));
+		act->setCustomEngineActionEvent(AGOSAction::kActionToggleSubtitle);
+		act->addDefaultInputMapping("t");
+		gameKeyMap->addAction(act);
+
+		act = new Action("TOGGLESPEECH", _("Switches between subtitles only and combined speech and subtitles"));
+		act->setCustomEngineActionEvent(AGOSAction::kActionToggleSpeech);
+		act->addDefaultInputMapping("v");
+		gameKeyMap->addAction(act);
+	}
+
+	if (gameId == "swampy" ||
+		gameId == "puzzle" ||
+		gameId == "jumble") {
+		act = new Action("HIGHSPEED", _("High speed mode on/off in Swampy Adventures"));
+		act->setCustomEngineActionEvent(AGOSAction::kActionSpeed_GTYPEPP);
+		act->addDefaultInputMapping("F12");
+		gameKeyMap->addAction(act);
+	}
+
+	act = new Action("KEYYES", _("Press Yes Key"));
+	act->setCustomEngineActionEvent(AGOSAction::kActionKeyYes);
+	act->addDefaultInputMapping("JOY_LEFT_SHOULDER");
+	gameKeyMap->addAction(act);
+
+	act = new Action("KEYNO", _("Press No Key"));
+	act->setCustomEngineActionEvent(AGOSAction::kActionKeyNo);
+	act->addDefaultInputMapping("JOY_RIGHT_SHOULDER");
+	gameKeyMap->addAction(act);
+
+
+	const Common::HardwareInputTableEntry *JoyEntry = Common::defaultJoystickButtons;
+	act = new Action("PRESSANYKEY", _("Press Any Key"));
```

Comment:
I am fairly certain this is the wrong path to go down. If you already need to disable all other keymaps, why not just disable all keymaps and track a bool for any key or joy pressed? It's going to be odd to open the keymap options to see this with literally every joy entry mapped to it.

Reply:
Acknowledged